message-format.md

```diff
@@ -1,47 +1,106 @@
+| int_map     | `{"answer": 42}`                                         |                                                                                                                             |
+| str_map     | `{"ip": "192.168.1.1"}`                                  |                                                                                                                             |
+| float_map   | `{"pi": 3.14}`                                           |                                                                                                                             |
+| bool_map    | `{"enabled": true}`                                      |                                                                                                                             |
+| object      | `{"nested": {"objects": "supported"}}`                   | Used for complex objects which cannot be mapped to primitive types. Structure is defined uniquely per service or interface. |
+| bin         | `"U28gbG9uZywgYW5kIHRoYW5rcyBmb3IgYWxsIHRoZSBmaXNoLg=="` | Used for `base64` encoded binary payloads.                                                                                  |
+
+## Time Format
+
+Time must be formatted in accordance to the RFC3339 format. A message parser must be able to read all the following timestamps as valid.
+When producing messages it is recommended to only use the first one.
+
+| Format                                   | Recommended |
+|------------------------------------------|-------------|
+| `"2006-01-02T15:04:05.999999999Z07:00"`  | Yes         | 
+| `"2006-01-02T15:04:05.999999999Z0700"`   | No          | 
+| `"2006-01-02 15:04:05.999999999 Z0700"`  | No          | 
+| `"2006-01-02 15:04:05.999999999 Z07:00"` | No          | 
+
+## Interface Format
+
+Each interface consist of three segments separated by a dot in accordance to this schema `{type}.{attribute}.{action}`:
+
+| Segment     | Example      | Description                                                                                                                                                                         |
+|-------------|--------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| *type*      | `cmd`        | Defines the **type** of the interface, either `cmd` for consumed incoming commands or `evt` for produced outgoing events.                                                           |
+| *attribute* | `lvl`        | Defines the **attribute** or a **domain** of the service. A service might have multiple attributes, while each attribute might have multiple interfaces. Must utilize `snake_case`. |
```

Comment:
Has to utilize

Reply:
Again https://www.ietf.org/rfc/rfc2119.txt